src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -254,28 +254,28 @@ public override Task<T> WaitForExternalEvent<T>(string name)
 
             lock (this.pendingExternalEvents)
             {
-                object tcsRefList;
-                List<TaskCompletionSource<T>> tcsList;
+                ICollection tcsRefCollection;
+                Queue<TaskCompletionSource<T>> tcsQueue;

```

Comment:
Now that I know there is a non-generic `Queue` class, we should simply use that instead of `ICollection` or `Queue<T>`.

Reply:
Thank you! I will change it.